### PR TITLE
Reflect in the documentation that the LOHThreshold has to be less than region_size

### LINKED ISSUE #95219: Crash when allocating 4+ MB with high LOH threshold
### Description

I realized that there is no higher limit on `DOTNET_GCLOHThreshold` so I wondered how the GC would handle it if I tried to allocate 4+ MB with a LOH threshold that is bigger than that.

The application hangs with the memory usage continuously increasing. I didn't debug in detail, but it looks like the GC enters in an infinite loop where it tries to allocate a new 4MB basic region, only to find that the allocation doesn't fit in any available region.

### Reproduction Steps

Set `DOTNET_GCLOHThreshold=6400000` then run the code:

```csharp
var alloc = new byte[1024 * 1024 * 4];
```

### Expected behavior

Either of:
 - The array is successfully allocated in gen 0
 - The LOH threshold is capped at ~3 MB and the array is successfully allocated in LOH
 - The CLR refuses to start

### Actual behavior

The GC is stuck in a seemingly infinite loop, creating new regions.

### Regression?

The issue only happens with regions enabled.

### Known Workarounds

Reduce the LOH threshold or disable regions.

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by cshung**
## Summary

We recently discovered that we cannot let the user to allocate arbitrarily large objects in the small object heap, therefore we implement a change (https://github.com/dotnet/runtime/pull/95485) in the runtime and this change in the doc reflects that.

The actual cap is the `gc_region_size`. It is an implementation detail and it changes according to different user/machine configurations. Therefore, instead of specifying the maximum, we just said it could be capped. I am open to suggestion on how to make it sounds more clear.

The product code change fixes https://github.com/dotnet/runtime/issues/95219, and we don't have a docs issue for this.

@Maoni0, @dotnet/gc 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/runtime-config/garbage-collector.md](https://github.com/dotnet/docs/blob/a90b3470f6baf64a5278d1b68ce45628ed7d365a/docs/core/runtime-config/garbage-collector.md) | [Runtime configuration options for garbage collection](https://review.learn.microsoft.com/en-us/dotnet/core/runtime-config/garbage-collector?branch=pr-en-us-38763) |

### CODE REVIEW

File: docs/core/runtime-config/garbage-collector.md [hunk 1]
```diff
@@ -669,6 +669,7 @@ Project file:
 - Specifies the threshold size, in bytes, that causes objects to go on the large object heap (LOH).
 - The default threshold is 85,000 bytes.
 - The value you specify must be larger than the default threshold.
+- The value can be capped by the runtime to the maximum possible size for the current configuration.
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:672]: Should we add that this is in .NET 9 and later, or is that unnecessary?

**Review by gewarren — APPROVED** [review]

File: docs/core/runtime-config/garbage-collector.md (hunk 1 above)
Comment (cshung) [reply docs/core/runtime-config/garbage-collector.md:672]: I think it is unnecessary. In principle, the value should have been capped in .net 7/8 as well, we just don't have a business reason to fix those earlier versions, yet. We might if we find a reason to.

**Comment by Maoni0** [comment]
it'd be nice to mention how the user can call the config api to find out the actual value that's set.

File: docs/core/runtime-config/garbage-collector.md [hunk 2]
```diff
@@ -669,6 +669,7 @@ Project file:
 - Specifies the threshold size, in bytes, that causes objects to go on the large object heap (LOH).
 - The default threshold is 85,000 bytes.
 - The value you specify must be larger than the default threshold.
+- The value can be capped by the runtime to the maximum possible size for the current configuration. The value actually being used at runtime can be inspected through the [GC.GetConfigurationVariables](https://learn.microsoft.com/en-us/dotnet/api/system.gc.getconfigurationvariables) API.
```
Comment (gewarren) [inline docs/core/runtime-config/garbage-collector.md:672]:
```suggestion
- The value might be capped by the runtime to the maximum possible size for the current configuration. You can inspect the value in use at run time through the <xref:System.GC.GetConfigurationVariables?displayProperty=nameWithType> API.
```

**Pull request merged**